### PR TITLE
SM-706: Compress payload on send status, decompress on benthos ingest

### PR DESCRIPTION
**Pull request by zakharenkodmytro**
- added option to gzip Payload and send it over MQTT
- refactored code

### CODE REVIEW

File: internal/worker_runner.go [hunk 1]
```diff
@@ -33,15 +37,17 @@ type workerRunner struct {
...
 	sendPayloadInterval time.Duration
+	archive             bool
 }
 
 func NewWorkerRunner(unitID uuid.UUID, addr *common.Address, loggerSettingsSvc loggers.TemplateStore,
 	queueSvc queue.StorageQueue, dataSender network.DataSender, logger zerolog.Logger, fpRunner FingerprintRunner, pids *models.TemplatePIDs, settings *models.TemplateDeviceSettings) WorkerRunner {
 	signalsQueue := &SignalsQueue{lastTimeChecked: make(map[string]time.Time)}
 	// Interval for sending status payload to cloud. Status payload contains obd signals and non-obd signals.
 	interval := 20 * time.Second
+	archiveData := true
```
Comment (JamesReate) [inline internal/worker_runner.go:48]: i think this is alright, maybe we make it a config parameter later on

File: internal/worker_runner.go [hunk 2]
```diff
@@ -124,6 +140,32 @@ func (wr *workerRunner) Run() {
...
 }
 
+// DeviceStatusData is formatted as json, gzip compressed, then base64 compressed.
+// This is done to reduce the size of the payload sent to the cloud over MQTT.
+func compressDeviceStatusData(s models.DeviceStatusData) (*models.DeviceStatusCompressedData, error) {
+	payload, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+	println(string(payload))
```
Comment (JamesReate) [inline internal/worker_runner.go:150]: do we still need this println or was that for debug?

**Comment by JamesReate** [comment]
@zer0stars thoughts on this - basically we're compressing just the data portion of the payload (under a different name `payloadBase64`, open to naming suggestions). But we could also just compress the entire thing, although realizing that may make the benthos side harder.

File: internal/worker_runner.go (hunk 2 above)
Comment (zakharenkodmytro) [reply internal/worker_runner.go:150]: yep, it was just for debugging purposes

**Review by JamesReate — APPROVED** [review]

**Pull request merged**